lib/apiservers/engine/backends/filter/container.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filter
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/docker/engine-api/types"
+
+	"github.com/vmware/vic/lib/apiservers/engine/backends/cache"
+	"github.com/vmware/vic/lib/apiservers/portlayer/models"
+)
+
+// reused from docker/docker/daemon/list.go
+type ContainerListContext struct {
+	FilterContext
+
+	// idx is the container iteration index for this context
+	Counter int
+	// ExitCode for the passed container
+	ExitCode int
+	// exitAllowed is a list of exit codes allowed to filter with
+	exitAllowed []int
+	// ContainerListOptions is the filters set by the user
+	*types.ContainerListOptions
+}
+
+/*
+*	IncludeContainer will evaluate the filter criteria in listContext against the provided
+* 	container and determine what action to take.  There are three options:
+*		* Include
+*		* Exclude
+*		* Stop
+*
+ */
+func IncludeContainer(listContext *ContainerListContext, container *models.ContainerInfo) FilterAction {
+
+	// filter common requirements
+	act := filterCommon(&listContext.FilterContext)
+	if act != Include {
```

Comment:
had it like that in an earlier iteration, but since the type is FilterAction I see the Action suffix as too verbose.....I'm fine with it and will change it to include the Action suffix